src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -169,43 +169,58 @@ private void processZipFileIfEligible(CloudBlobContainer container, String zipFi
             logAbortedProcessingNonExistingFile(zipFilename, container.getName());
         } else {
             cloudBlockBlob.downloadAttributes();
-            processZipFile(container, cloudBlockBlob, zipFilename);
+            leaseAndProcessZipFile(container, cloudBlockBlob, zipFilename);
         }
     }
 
-    private void processZipFile(
+    private void leaseAndProcessZipFile(
         CloudBlobContainer container,
         CloudBlockBlob cloudBlockBlob,
         String zipFilename
     ) throws StorageException, IOException {
-        Optional<String> leaseId = blobManager.acquireLease(cloudBlockBlob, container.getName(), zipFilename);
+        Optional<String> leaseIdOption = blobManager.acquireLease(cloudBlockBlob, container.getName(), zipFilename);
 
-        if (leaseId.isPresent()) {
-            Envelope envelope = envelopeProcessor.getEnvelopeByFileAndContainer(container.getName(), zipFilename);
+        if (leaseIdOption.isPresent()) {
+            String leaseId = leaseIdOption.get();
```

Comment:
worth logging this lease id when acquired and when released :)

Reply:
done